### PR TITLE
404 page

### LINKED ISSUE #356: 404 error page
We should have an error page so we can track possible link errors or stuff people may be looking for.

### TODO

- [x] add 404 page component
- [x] design 404 page component
- [x] handle transitions to unknown path on client side
- [x] respond with 404 page from server for unknown paths (@marcoow: we currently pre-render the 404 page as `/404/index.html` - setting that as the default response for 404 errors should only be a server side setting)

### PR DESCRIPTION
**Pull request by marcoow**
This is 50% done - missing design and we need to configure the server so that it responds with the pre-rendered 404 page on 404 errors.

@ghislaineguerin, @pichfl: can one of you design the page component that I added (pushing to this branch)?

closes #356

### CODE REVIEW

**Comment by ghislaineguerin** [comment]
@marcoow @pichfl 
https://share.goabstract.com/9433e105-1764-4800-b5a0-924803e081b4?mode=design&sha=dcdddd095a9faff8e0ce0a112c0a3b270f668f26
Here's the 404 page design. I have included some links so that hopefully there's some navigation back home or to other pages.

image

**Comment by marcoow** [comment]
@pichfl can you add the HTML/CSS to this branch?

**Review by ghislaineguerin — APPROVED** [review]
👍

File: src/ui/components/Page404/template.hbs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    </h2>
+    <ul class="typography.lead">
+      <li class="elsewhere">
+        <a class="typography.link" href="/" data-internal>
+          Return to Homepage
+        </a>
+      </li>
+      <li class="elsewhere">
+        <a class="typography.link" href="/about" data-internal>
+          About Simplabs
```
Comment (marcoow) [inline src/ui/components/Page404/template.hbs:28]: lowercase please 🙏

File: src/ui/components/Page404/template.hbs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        </a>
+      </li>
+      <li class="elsewhere">
+        <a class="typography.link" href="/blog" data-internal>
+          Simplabs Blog
```
Comment (marcoow) [inline src/ui/components/Page404/template.hbs:33]: see above - lowercase please 🙏

**Pull request merged**